pilot/pkg/config/kube/gateway/inferencepool.go

```diff
@@ -0,0 +1,290 @@
+	// 	r.Logger.Errorf(retErr.Error())
+	// 	r.Metrics.IncrementReconcilerErrors("Reconciler.processInferencePool", r.reconcilerType)
+	// 	return retErr
+	// }
+
+	shadowSvc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      svcName,
+			Namespace: pool.GetNamespace(),
+		},
+	}
+	if shadowSvc.Labels == nil {
+		shadowSvc.Labels = map[string]string{}
+	}
+	shadowSvc.Labels[InferencePoolRefLabel] = pool.GetName()
+	//TODO: for now, supporting only service
+	extRef := pool.Spec.ExtensionRef
+	shadowSvc.Labels[InferencePoolExtensionRefSvc] = string(extRef.Name)
+	if extRef.PortNumber != nil {
+		shadowSvc.Labels[InferencePoolExtensionRefPort] = strconv.Itoa(int(*extRef.PortNumber))
+	} else {
+		shadowSvc.Labels[InferencePoolExtensionRefPort] = "9002"
+	}
+	shadowSvc.Labels["internal.istio.io/service-semantics"] = "inferencepool"
+	shadowSvc.Spec.Ports = []corev1.ServicePort{
+		{
+			Protocol:   "TCP",
+			Port:       int32(54321),
+			TargetPort: intstr.FromInt(int(pool.Spec.TargetPortNumber)),
+		},
+	}
+	if shadowSvc.Spec.Selector == nil {
+		shadowSvc.Spec.Selector = map[string]string{}
+	}
+	for k, v := range pool.Spec.Selector {
+		shadowSvc.Spec.Selector[string(k)] = string(v)
+	}
+	shadowSvc.Spec.ClusterIP = corev1.ClusterIPNone
+	gvk := pool.GetObjectKind().GroupVersionKind()
+	log.Infof("LIOR111: %v", gvk)
+	shadowSvc.SetOwnerReferences([]metav1.OwnerReference{
+		{
+			// APIVersion: pool.APIVersion,
+			APIVersion: "inference.networking.x-k8s.io/v1alpha2",
+			Kind:       "InferencePool",
+			Name:       pool.GetName(),
+			UID:        pool.GetUID(),
+		},
+	})
+	return shadowSvc, nil
+}
+
+// ensureLabels adds management labels to resources
+func ensureLabels(obj metav1.Object, poolName string) {
+	labels := obj.GetLabels()
+	if labels == nil {
+		labels = make(map[string]string)
+	}
+	labels[InferencePoolRefLabel] = poolName
+	obj.SetLabels(labels)
+}
+
+func (ic *InferencePoolController) apply(obj interface{}) error {
+	if service, ok := obj.(*corev1.Service); ok {
+		name := service.Name
+		namespace := service.Namespace
+		// Get the existing service if it exists
+		existingService := ic.services.Get(service.Name, service.Namespace)
+
+		// Check if we can manage this service
+		if existingService != nil {
+			canManage, _ := ic.canManage(existingService)
+			if !canManage {
+				log.Debugf("skipping service %s/%s, already managed by another controller", namespace, name)
+				return nil
+			}
+		}
+
+		var err error
+		if existingService == nil {
+			// Create the service if it doesn't exist
+			_, err = ic.client.Kube().CoreV1().Services(service.Namespace).Create(
+				context.Background(), service, metav1.CreateOptions{})
+		} else {
+			// Update the service if it exists
+			// Note: We need to use Update instead of Patch for simpler implementation
+			// This means we might overwrite other fields, but that's the expected behavior
```

Comment:
Note a variety of controllers add arbitrary labels/annotations to services after creation so this may be problematic. I don't mind for short term of course, especially as I hope this is temporary.